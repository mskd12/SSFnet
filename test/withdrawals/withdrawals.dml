# 
# withdrawals.dml
#
# Two ASes, each with one router.  The routers are directly connected,
# and are each running BGP.
#
#        router          router
# 
#        1:1(1) }------{ 2:1(1)
#

_schema [ _find .schemas.Net ]

Net [
  
  frequency 1000000000   # nanosecond simulation frequency

  bgpoptions [
    validation_test 6      # prints messages associated with this test
  ]

  # Nets are essentially identical except that the first one is running a
  # pseudo-protocol to assist with validation.
  Net [
    id 1
    AS_status boundary
    router [
      id 1
      graph [
        ProtocolSession [
          name test  use SSF.OS.BGP4.test.withdrawals.src.WithdrawalsTester
        ]
        ProtocolSession [
          name bgp  use  SSF.OS.BGP4.BGPSession
          autoconfig false
          connretry_time 120  min_as_orig_time 15
          reflector false
          neighbor [ as 2  address 1(1)  use_return_address 1(1)
                     _extends .basic_ebgp_neighbor ]
        ]
        ProtocolSession [ name socket use SSF.OS.Socket.socketMaster ]
        ProtocolSession [ name tcp    use SSF.OS.TCP.tcpSessionMaster ]
        ProtocolSession [ name ip     use SSF.OS.IP ]
      ]
      interface [ id 0  virtual true ]
      interface [ id 1 ]
    ]
  ]

  Net [
    id 2
    AS_status boundary
    router [
      id 1
      graph [
        ProtocolSession [
          name bgp  use  SSF.OS.BGP4.BGPSession
          autoconfig false
          connretry_time 120  min_as_orig_time 15
          reflector false
          neighbor [ as 1  address 1(1)  use_return_address 1(1)
                     _extends .basic_ebgp_neighbor ]
        ]
        ProtocolSession [ name socket use SSF.OS.Socket.socketMaster ]
        ProtocolSession [ name tcp    use SSF.OS.TCP.tcpSessionMaster ]
        ProtocolSession [ name ip     use SSF.OS.IP ]
      ]
      interface [ id 0  virtual true ]
      interface [ id 1 ]
    ]
  ]

  link [ attach 1:1(1) attach 2:1(1) delay 0.03 ]
]
