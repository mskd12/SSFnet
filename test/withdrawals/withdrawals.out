~#
~# = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
~# WITHDRAWALS Validation Test: successful if six steps printed below
~# = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
~#
~# 1. AS X: bogus route being advertised to AS Y
~# 2. AS Y: received advertisement from AS X
~# 3. AS Y: bogus route added to routing table
~# 4. AS X: withdrawal for bogus route being sent to AS Y
~# 5. AS Y: received withdrawal from AS X
~# 6. AS Y: bogus route removed from forwarding table
~#             *** TEST SUCCESSFUL ***
~#
